cmd/config/internal/commands/cmdlistsetters.go

```diff
@@ -147,6 +162,7 @@ func (r *ListSettersRunner) ListSubstitutions(c *cobra.Command, args []string) e
 		return nil
 	}
 	table.Render()
+	r.List.Substitutions = r.List.Substitutions[:0]
```

Comment:
Can we add a comment to explain why this is needed?

Reply:
Done.